### PR TITLE
Feat: Tags CRUD View + Tests; Closes #1120

### LINKED ISSUE #1120: Tags CRUD view completion
There are basic CRUD views already I think, but a create view is missing. Currently Tags are created directly by adding them via the quest or badge form.

Also missing is an edit view. There should be the ability to edit the name of a tag from the lost view (just like all our other CRUD lost views)

Finally a delete tag button with confirmation page (confirmation page lists all items that will lose the tag of deleted, similar to tag detail view)

Admin buttons on right of the list view table, just like other list views.

### PR DESCRIPTION
**Pull request by 9921622**
Tags are accessed from the admin menu

create/update:
image

detail view:
image

delete view:
image

### CODE REVIEW

File: src/tags/templates/tags/detail.html [hunk 1]
```diff
@@ -9,24 +9,32 @@
...
+<br>
+
+<div class="panel panel-default">
+    <div class="panel-heading"><h4>Quests ({{ quests|length }}) <i class="fa fa-shield fa-fw"></i></h4></div>
+    <div class="panel-body">
+        <p></p>
+        <ul>
+            {% for quest in quests %}
+                <li><a href="{{ quest.get_absolute_url }}">{{ quest }}</a></li>
+            {% endfor %}
```
Comment (tylerecouture) [inline src/tags/templates/tags/detail.html:21]: Missing the empty case (see deleted code)

File: src/tags/templates/tags/detail.html [hunk 2]
```diff
@@ -9,24 +9,32 @@
...
+
+
+<div class="panel panel-default">
+    <div class="panel-heading"><h4>Badges ({{ badges|length }}) <i class="fa fa-certificate fa-fw"></i></h4></div>
+    <div class="panel-body">
+        <p></p>
+        <ul>
+            {% for badge in badges %}
+                <li><a href="{{ badge.get_absolute_url }}">{{ badge }}</a></li>
+            {% endfor %}
```
Comment (tylerecouture) [inline src/tags/templates/tags/detail.html:34]: Empty case

File: src/tags/forms.py [hunk 3]
```diff
@@ -14,3 +19,23 @@ class Media:
...
+class TagForm(forms.ModelForm):
+
+    class Meta:
+        model = Tag
+        fields = '__all__'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields['slug'].required = False
```
Comment (tylerecouture) [inline src/tags/forms.py:33]: What's the purpose of adding the slug field? Can we remove it? I don't think the user needs to know about it at all, if it's required for the form, can we just hide it?

File: src/tags/views.py [hunk 4]
```diff
@@ -33,48 +38,57 @@
...
 
     def get_context_data(self, **kwargs):
-        kwargs['quests'] = Quest.objects.filter(tags__name=self.object.name)
-        kwargs['badges'] = Badge.objects.filter(tags__name=self.object.name)
+        ctx = super().get_context_data(**kwargs).copy()
+        ctx['quests'] = Quest.objects.filter(tags__id=self.object.id)
+        ctx['badges'] = Badge.objects.filter(tags__id=self.object.id)
+        return ctx
 
-        return super().get_context_data(**kwargs)
```
Comment (tylerecouture) [inline src/tags/views.py:48]: How come you changed the structure of this?

**Review by tylerecouture — CHANGES_REQUESTED** [review]
A few minor comments. Also:

 from the screenshots, the styling doesn't match between the detail vs delete view, can you make them match? Perhaps put that code into a template snippet so you can call common code from both instead of duplicating code. (Keep it DRY)

File: src/tags/forms.py (hunk 3 above)
Comment (9921622) [reply src/tags/forms.py:33]: its set as unique=True and non optional (at least in the github version)
image
ill remove it from the form, but set it in the save func
image

File: src/tags/views.py (hunk 4 above)
Comment (9921622) [reply src/tags/views.py:48]: i actually have no idea, reverted

**Comment by 9921622** [comment]
added requested changes

detail view
image
delete view
image


update view without slug field
image

**Comment by 9921622** [comment]
forgot to link this to issue #1120

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Awesome!  Two small things:

1. For consistancy across the other apps, can you call the directory "snippets" instead of "partials"?

2. Can you add a quick test to check that the admin buttons aren't displayed to non-staff users?

**Comment by 9921622** [comment]
Added requested changes

File: src/tags/views.py [hunk 5]
```diff
@@ -33,48 +38,56 @@
...
 
-# @method_decorator(staff_member_required, name='dispatch')
-# class TagDelete(NonPublicOnlyViewMixin, DeleteView):
-#     model = MenuItem
-#     success_url = reverse_lazy('utilities:menu_items')
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs).copy()
+        ctx['quests'] = Quest.objects.filter(tags__id=self.object.id)
+        ctx['badges'] = Badge.objects.filter(tags__id=self.object.id)
+        return ctx
```
Comment (tylerecouture) [inline src/tags/views.py:93]: this one still has the alternate implementation.  doesn't really matter, but might as well be consistent?

**Review by tylerecouture — APPROVED** [review]
Still one inconsistent context_data_method, but I'll approve.

File: src/tags/views.py (hunk 5 above)
Comment (9921622) [reply src/tags/views.py:93]: doesnt really matter since my next pr changes this

**Pull request merged**